status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,372 @@
+  optional bool ephemeral = 31;
+}
+```
+
+- `payload` MUST be set to the full encrypted payload received from the higher layers
+- `version` MUST be set to `1`
+- `ephemeral` MUST be set to `true` if the app-level message is ephemeral
+- `content_topic` MUST be set to the app-level content topic
+- `timestamp` MUST be set to the current Unix epoch timestamp (in nanosecond precision)
+
+### Pubsub topics and sharding
+
+All Waku messages are published to pubsub topics as defined in [23/WAKU2-TOPICS](../../waku/informational/23/topics.md).
+Since pubsub topics define a routing layer for messages,
+they can be used to shard traffic.
+The pubsub topic used for publishing a message depends on the app-level [functional scope](#functional-scope).
+
+**Self-addressed messages**
+
+The application MUST define at least one distinct pubsub topic for self-addressed messages.
+The application MAY defined a set of more than one pubsub topic for self-addressed messages to allow traffic sharding for scalability.
+
+**Global messages**
+
+The application MUST define at least one distinct pubsub topic for global control messages and global content messages.
+The application MAY defined a set of more than one pubsub topic for global messages to allow traffic sharding for scalability.
+It is RECOMMENDED that separate pubsub topics be used for global control messages and global content messages.
+
+**Community messages**
+
+The application SHOULD define at least one separate pubsub topic for each separate community's community control and community content messages.
+The application MAY define a set of more than one pubsub topic per community to allow traffic sharding for scalability.
+It is RECOMMENDED that separate pubsub topics be used for global control messages and global content messages.
+
+**Large messages**
+
+The application MAY define separate pubsub topics for large messages.
+These pubsub topics for large messages MAY be distinct for each functional scope.
+
+### Resource usage
+
+The application can use a range of Waku protocols to interact with the Waku transport layer.
+The specific set of Waku protocols used depend on desired functionality and resource usage profile for the specific client.
+Resources can be restricted in terms of bandwidth and computing resources.
+
+Waku protocols that are more appropriate for resource-restricted environments are often termed "light protocols".
+Waku protocols that consume more resources, but simultaneously contribute more to Waku infrastructure, are often termed "full protocols".
+The terms "full" and "light" is just a useful abstraction than a strict binary, though,
+and Status clients can operate along a continuum of resource usage profiles,
+each using the combination of "full" and "light" protocols most appropriate to match its environment and motivations.
+
+To simplify interaction with the selection of "full" and "light" protocols,
+Status clients MUST define a "full mode" and "light mode"
```

Comment:
```suggestion
Status clients MUST define a "full mode" and "light mode"
```
Same, I want to potentially propose alternative terminology. Will work on it and share draft.